app/Http/Controllers/RoomController.php

```diff
@@ -199,37 +176,32 @@ public function update($roomId, Request $request)
         }
 
         $minRating = $request->get('rated')
-            ? $request->get('min_rating')
+            ? (int)$request->get('min_rating')
             : 0;
 
         $room->goal()->associate(Goal::whereIdentifier($request->get('goal'))->first());
         $room->update([
             'description' => $request->get('description'),
-            'is_rated' => $request->get('rated'),
-            'is_anonymous' => $request->get('anonymity'),
-            'max_players' => $request->get('max_players'),
+            'is_rated' => (bool)$request->get('rated'),
+            'is_anonymous' => (bool)$request->get('anonymity'),
+            'max_players' => (int)$request->get('max_players'),
             'min_rating' => $minRating,
-            'map' => $request->get('map'),
-            'seed' => Carbon::now()->unix(),
+            'map' => (int)$request->get('map'),
         ]);
 
-        return new Response($room);
+        return new Response();
     }
 
     /**
      * Destroy a room.
      * TODO: We should decide when a user should be able to delete a room precisely
      *
-     * @param $roomId
+     * @param Room $room
      * @return Response
-     * @throws \Exception
+     * @throws ValidationException|Exception
```

Comment:
I think the user should only be able to destroy their lobby before it has officially started. Once the user clicks "Start early" or the max number of players join their game, the user shouldn't be able to destroy the game.

Reply:
Also I think the game shouldn't start as soon as the last player joins. There should be a period of time where the last player can "check out" the lobby, and potentially leave.

Reply:
Well we're updating the game to never show the map to anyone before the game starts so I think this is a non-issue. When you join a casual lobby you will be able to see all the game configuration options and other players in the lobby so I think starting as soon as the last player is in the game is fine.

Reply:
When did we decide on this? XD I think it should be an option rather than forced. Because otherwise what about custom map designs?

Reply:
All of these customization options are turning into scope creep. For now we will not be showing the map before the game starts. Players also won't be able to create custom maps in the first release. Future releases might have custom maps if we have time but that is not a priority right now. If we add custom maps in the future, we can have a 'preview' or something before they join.

This thread is getting off-topic from the code in question here. Regardless of custom maps, the user should only be able to destroy their lobby while it is in the 'open' state.

Reply:
I think the best way to go would be to "start" the game instantly when the room fulfills the requirements (otherwise we'd need to have some cronjob checking for rooms to start after a countdown or something strange like that), setting `started_at` to the current timestamp as soon as it is started early or when it reaches max players.

If we want some sort of warm-up time we can always add some validation later that no events may be processed during the first _n_ hours of the game, which would be a simple addition. I'd suggest to put it in an issue so we can debate the feature there.